### PR TITLE
feat: add support of order and limit on Repository::getBy

### PR DESCRIPTION
**Pull request by vgreb**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Licence       | Apache-2.0
| Fixed tickets |

### CODE REVIEW

File: src/Ting/Repository/Repository.php [hunk 1]
```diff
@@ -289,10 +289,12 @@ public function getAll($forceMaster = false)
      * @param bool  $forceMaster
      * @return CollectionInterface<T>
      */
-    public function getBy(array $criteria, $forceMaster = false)
+    public function getBy(array $criteria, $forceMaster = false, $order = [], $limit = 0)
```
Comment (xavierleune) [inline src/Ting/Repository/Repository.php:292]: can you add the relevant type hints in the signature ?

File: src/Ting/Repository/Metadata.php [hunk 2]
```diff
@@ -584,6 +584,28 @@ public function getByCriteria(
         return $queryGenerator->getByCriteria($criteriaColumn, $collectionFactory, $forceMaster);
     }
 
+    public function getByCriteriaWithOrderAndLimit(
+        array $criteria,
+        $orderBy,
```
Comment (xavierleune) [inline src/Ting/Repository/Metadata.php:589]: can you add the relevant type hints in the signature ?

**Review by xavierleune — CHANGES_REQUESTED** [review]

**Comment by xavierleune** [comment]
Can this change be merged in 3.x instead ?

File: src/Ting/Repository/Metadata.php (hunk 2 above)
Comment (vgreb) [reply src/Ting/Repository/Metadata.php]: I added typehint for parameters `$order`,  `$limit` and `$forceMaster`

File: src/Ting/Repository/Repository.php (hunk 1 above)
Comment (vgreb) [reply src/Ting/Repository/Repository.php]: I added typehint for parameters `$order` and `$limit`

**Comment by vgreb** [comment]
> Can this change be merged in 3.x instead ?

I don't see any breaking changes, I think this could be merged in 3.X.

**Pull request closed without merging**